skiko/src/awtMain/kotlin/org/jetbrains/skiko/windows/JumpList.kt

```diff
@@ -0,0 +1,156 @@
+package org.jetbrains.skiko.windows
+
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import org.jetbrains.skiko.hostOs
+
+/**
+ * Provides access to Windows Jump List features
+ */
+object JumpList {
+    /**
+     * Returns a value that indicates whether Jump Lists are supported.
+     */
+    fun isSupported(): Boolean = when {
+        hostOs.isWindows -> true
+        else -> false
+    }
+
+    /**
+     * Wraps a single Jump List building transaction.
+     */
+    fun build(block: JumpListBuilder.() -> Unit) = when {
+        hostOs.isWindows -> JumpListBuilder().use { builder ->
+            builder.initialize()
+            block(builder)
+            builder.commit()
+        }
+        else -> error("Jump List is only supported on Windows")
+    }
+}
+
+/**
+ * Represents an item in the Jump List.
+ */
+data class JumpListItem(val title: String, val arguments: String) {
+    var attributes: JumpListItemAttributes? = null
+}
+
+/**
+ * Represents optional attributes that can be added to a Jump List item.
+ */
+class JumpListItemAttributes {
+    var description: String? = null
+    var icon: JumpListItemIcon? = null
+}
+
+/**
+ * Represents an icon that can be assigned to a Jump List item.
+ */
+data class JumpListItemIcon(val path: String, val index: Int) {
+    companion object {
+        internal fun fromParts(path: String?, num: Int): JumpListItemIcon? {
+            if (path.isNullOrBlank()) return null
```

Comment:
Should be `isNullOrEmpty()` here for consistency with `JumpListInteropItem.fromInterop()`.